core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java

```diff
@@ -253,12 +257,28 @@ private void updateStates(Stack stack, Collection<InstanceMetaData> failedInstan
         clusterService.updateClusterCertExpirationState(stack.getCluster(), hostCertExpiring);
         clusterOperationService.reportHealthChange(stack.getResourceCrn(), newFailedNodeNamesWithReason, newHealthyHostNames);
         if (!failedInstances.isEmpty()) {
-            clusterService.updateClusterStatusByStackId(stack.getId(), DetailedStackStatus.NODE_FAILURE);
+            String accountId = Crn.safeFromString(stack.getResourceCrn()).getAccountId();
+            if (!stopStartEnabled(accountId, stack.getCloudPlatform())) {
+                clusterService.updateClusterStatusByStackId(stack.getId(), DetailedStackStatus.NODE_FAILURE);
```

Comment:
Should we be introducing a new Status in com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status (i think this is stack/cluster status). Something along the lines of AVAILABLE_WITH_STOPPED_INSTANCES.

This would not qualify via isAvailable(), isStopState. It would via isStartState (not sure if this will be problematic). Stop-Start scaling would explicitly ALLOW this state, and other operations allowed during stop-start scaling would be enhanced to check for this state as an alternate.

Haven't seen the detailed usage of states... so could be off here. Even with this change, a similar check will be required somewhere, correct?

@lnardai, @sodre90 - any recommendations on a good way to handle states here? If there are STOPPED instances - certain operations like STOP CLUSTER, REPAIR need to be allowed. However, others like UPGRADE etc need to be blocked for now. Also, with STOPPED instances - the user should not see the cluster state as NODE_FAILURE (It should be AVAILABLE or a variant of AVAILABLE)

Reply:
I think this approach seems to work. But there are a couple of hurdles,

1. UI needs to be fixed for showing user-readable messages and showing stop button for this state.
2. "Autoscaling Trigger Failed. Autoscaling metrics collection from cluster failed.". Need to investigate this.
3. The status changes to Running after some time, need to investigate this.

<img width="1416" alt="Screen Shot 2022-01-13 at 6 34 32 PM" src="https://user-images.githubusercontent.com/18580046/149441890-35c7348c-fd75-4e36-a915-b6596aae1944.png">

Reply:
> I think this approach seems to work. But there are a couple of hurdles,
> 
> 1. UI needs to be fixed for showing user-readable messages and showing stop button for this state.
UI needs to be fixed for showing user-readable messages ?
> 2. "Autoscaling Trigger Failed. Autoscaling metrics collection from cluster failed.". Need to investigate this.
Would be strange if this were related. Can see AutoScaling being disabled ... failing to fetch metrics is strange though.
> 3. The status changes to Running after some time, need to investigate this.
This might be the regular sync task. Surprised it moved to Running instead of the status which indicates Failed Instances (if there were STOPPED instances).

Reply:
Please see the screenshot. The status should be "Available (there are stopped instances)", instead of printing the enum as is.

Reply:
1. Fixed here https://github.com/hortonworks/hortonworks-cloud/pull/3045
2. Need to build and deploy autoscale also to add the new status
3. Fixed it. There is a stack sync service that needed an update about the new status.